### PR TITLE
fixed: in empty dir it shows message, and context menu is enabled. solves #113

### PR DESCRIPTION
**Pull request by ashutosh7i**
i have solved this issue-
https://github.com/cockpit-project/cockpit-navigator/issues/113

GIF of progress-
Screencast from 04-12-23 06 43 07 PM IST

=> My changes
1. It checks for directory contents beforehand, and works accordingly, if dir is empty it sets the space to 200px with a message "Directory is Empty" for user.
2. The context menu can be envoked from anywhere in the area.
3. works in grid as well as other mode.

### CODE REVIEW

File: src/navigatorContextMenu.jsx [hunk 1]
```diff
@@ -47,11 +47,21 @@ export const ContextMenu = ({ parentId, contextMenuItems, setSelectedContext })
         };
 
         const parent = document.getElementById(parentId);
-        parent.addEventListener("contextmenu", _handleContextMenu);
-        document.addEventListener("click", _handleClick);
-
+        const isEmptyDirectory = parent === null;
+        // Check if the parent is present and the directory is not empty
+        if (parent && !isEmptyDirectory) {
```
Comment (jelly) [inline src/navigatorContextMenu.jsx:52]: This check does not make a ton of sense as it checks the same variable twice.

File: src/app.jsx [hunk 2]
```diff
@@ -607,29 +608,77 @@ const NavigatorCardBody = ({
         );
     };
 
+    const EmptyDirectoryMessage = () => {
+        return (
+            <EmptyState>
+                <EmptyStateBody>
+                    Directory is empty
```
Comment (jelly) [inline src/app.jsx:615]: This should be `{ _("Directory is empty")  }` as it needs to be translated.

File: src/app.jsx [hunk 3]
```diff
@@ -607,29 +608,77 @@ const NavigatorCardBody = ({
...
+                            : (
+                                <ListingTable
+                                  id="folder-view"
+                                  className="pf-m-no-border-rows"
+                                  variant="compact"
+                                  columns={[_("Name")]}
+                                  rows={sortedFiles.map(file => (
+                                      { columns: [{ title: <Item file={file} key={file.name} /> }] }))}
+                                />
+                            )
```
Comment (jelly) [inline src/app.jsx:651]: This makes not a lot of sense, it is already behind a `if (isGrid)` on line 627.

File: src/app.jsx [hunk 4]
```diff
@@ -607,29 +608,77 @@ const NavigatorCardBody = ({
...
+                    )}
+            </CardBody>
```
Comment (jelly) [inline src/app.jsx:681]: Same as above, isGrid is always false here

**Review by jelly — CHANGES_REQUESTED** [review]
Sorry for the very long delay, I've posted some review comments.

**Comment by garrett** [comment]
I commented about this @ 
- https://github.com/cockpit-project/cockpit-navigator/issues/113#issuecomment-1840520137
- https://github.com/cockpit-project/cockpit-navigator/issues/113#issuecomment-1842533288

I think this PR, when it lands, would close #113.

**Comment by ashutosh7i** [comment]
Sorry for a long delay from my side, i went to a hackathon (won it though😁) now i am working on changes.

**Comment by jelly** [comment]
Sorry this has been meanwhile fixed in https://github.com/cockpit-project/cockpit-navigator/commit/fc8a58ffec22cd2b713e0a609c5bdabfe4762be1

**Pull request closed without merging**